django/core/files/storage/filesystem.py

```diff
@@ -21,21 +24,19 @@ class FileSystemStorage(Storage, StorageSettingsMixin):
     Standard filesystem storage
     """
 
-    # The combination of O_CREAT and O_EXCL makes os.open() raise OSError if
-    # the file already exists before it's opened.
-    OS_OPEN_FLAGS = os.O_WRONLY | os.O_CREAT | os.O_EXCL | getattr(os, "O_BINARY", 0)
-
     def __init__(
         self,
         location=None,
         base_url=None,
         file_permissions_mode=None,
         directory_permissions_mode=None,
+        allow_overwrite=False,
```

Comment:
We need an update to the docs for [`FileSystemStorage`](https://docs.djangoproject.com/en/5.0/ref/files/storage/#the-filesystemstorage-class) adding `allow_overwite` and a `.. versionchanged::` note.
We also need a release note for `allow_overwite` and a deprecation/removed note for `OS_OPEN_FLAGS`.